### PR TITLE
Config manager transient support

### LINKED ISSUE #4442: Fields with 'transient' keyword should be ignored by ConfigManager
Example:
`public int valueThatWillBePartOfConfig = 393`
`public transient int valueThatWontBePartOfConfig = 854`

Only alternative currently is to have the field private, and make a public getter for it. But I feel like supporting transient fields would be better, since Gson does this too, and it shouldn't be too hard.

### PR DESCRIPTION
**Pull request by mainrs**
Resolves #4442 .

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,6 @@ changelog.txt
 /patches_old_bak/
 *.py
 *.7z
+
+# OS specific files
+.DS_STORE
```
Comment (diesieben07) [inline .gitignore]: Something like this belongs in your local, global `.gitignore`.

File: README.md [hunk 2]
```diff
@@ -49,14 +49,14 @@ Here is a short video from @cpw showing how to install and setup Forge
...
- the 1.10.x branch for Minecraft 1.10.2.
+ the 1.12.x branch for Minecraft 1.12.2.
 
 If you intend to modify a minecraft patch, or add a new one, always run
 ```gradlew genPatches``` to generate your patch and validate that you
 are not doing anything unexpected in your patch.
 
 ### Contributor License Agreement
-As stated in the (https://github.com/MinecraftForge/MinecraftForge/blob/1.10.x/LICENSE-new.txt)
+As stated in the (https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/LICENSE-new.txt)
```
Comment (diesieben07) [inline README.md]: You should only resolve one issue per PR.

File: .gitignore (hunk 1 above)
Comment (mainrs) [reply .gitignore]: Should I just remove the commit then?

File: README.md (hunk 2 above)
Comment (mainrs) [reply README.md]: I thought that doing one PR might be overkill for just changing some numbers. I'll split them up.

Comment (diesieben07) [reply README.md]: See https://github.com/MinecraftForge/MinecraftForge/issues/4618.

File: .gitignore (hunk 1 above)
Comment (diesieben07) [reply .gitignore]: Or revert this change.

**Comment by mainrs** [comment]
Hopefully I did that right. Not that of a git expert to be honest. Reverted + squashed.

**Review by diesieben07 — APPROVED** [review]

**Comment by liach** [comment]
The worst thing is that you could totally do it on GitHub (there has been a few pulls merged despite that they are totally edited on GitHub)

**Comment by liach** [comment]
Actually, I believe that a new annotation may work better than `transient` modifiers.

**Comment by mainrs** [comment]
@liach something along the lines of `@Ignore`?
Maybe allow both methods, add an annotation (like a lot of JSON serializers do) and allow the transient modifier. 
> transient is a Java keyword which marks a member variable not to be serialized when it is persisted to streams of bytes. When an object is transferred through the network, the object needs to be 'serialized'

This explanation is actually the exact use case here. Omitting fields for persistence. But I am fine with an annotation too.

**Comment by mainrs** [comment]
Re-opening since I kind of messed up my local repository.

**Pull request closed without merging**